pkg/tfshim/sdk-v2/schema.go

```diff
@@ -70,6 +71,11 @@ func (s v2Schema) DefaultValue() (interface{}, error) {
 	return withPatchedDefaults(s.tf).DefaultValue()
 }
 
+func (s v2Schema) HasDefault() bool {
+	sch := withPatchedDefaults(s.tf)
```

Comment:
Ohh good memories. Surprisingly it removes invalid s.Default == nil but not invalid s.DefaultFunc :/ That's fine, pre-existing condition, YAGNI, etc.